cockatrice/src/user_context_menu.cpp

```diff
@@ -94,6 +102,38 @@ void UserContextMenu::banUser_processUserInfoResponse(const Response &r)
     dlg->show();
 }
 
+void UserContextMenu::banUserHistory_processResponse(const Response &resp) {
+    const Response_BanHistory &response = resp.GetExtension(Response_BanHistory::ext);
+    if (resp.response_code() == Response::RespOk) {
+
+        if (response.ban_time_size() > 0) {
+            QTableWidget *table = new QTableWidget();
+            table->setWindowTitle(tr("Ban History"));
+            table->setRowCount(response.ban_time_size());
+            table->setColumnCount(5);
+            table->setSizePolicy(QSizePolicy::Expanding, QSizePolicy::Expanding);
+
+            table->setHorizontalHeaderLabels(
+                    QString(tr("Ban Time;Moderator;Ban Length;Ban Reason;Visible Reason")).split(";"));
+
+            for (int i = 0; i < response.ban_time_size(); ++i) {
+                table->setItem(i, 0, new QTableWidgetItem(QString::fromStdString(response.ban_time(i))));
+                table->setItem(i, 1, new QTableWidgetItem(QString::fromStdString(response.admin_name(i))));
+                table->setItem(i, 2, new QTableWidgetItem(QString::fromStdString(response.ban_length(i))));
+                table->setItem(i, 3, new QTableWidgetItem(QString::fromStdString(response.ban_reason(i))));
+                table->setItem(i, 4, new QTableWidgetItem(QString::fromStdString(response.visible_reason(i))));
+            }
+
+            table->resizeColumnsToContents();
+            table->setMinimumSize(table->horizontalHeader()->length() + (table->columnCount() * 5), table->verticalHeader()->length() + (table->rowCount() * 3));
+            table->show();
+        } else
+            QMessageBox::information(static_cast<QWidget *>(parent()), tr("Ban History"), tr("User has never been banned."));
```

Comment:
That is the way it works.  It just has an information box that pops up saying there is no ban history if there is none rather than showing an empty ban history table.